hack/run-e2e.sh

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The cert-manager Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+
+ROOT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )/.."
+cd "$ROOT_DIR"
+
+TMP_DIR=$(mktemp -d)
+trap "rm -rf $TMP_DIR" EXIT
+
+check_command() {
+  if ! [ -x "$(command -v $1)" ]; then
+    echo "Error: $1 is not installed." >&2
+    echo "Invoke script using 'nix develop -c ./hack/run.sh'" >&2
```

Comment:
Can we automatically invoke `nix develop -c ./hack/run-e2e.sh` if a user calls this script directly without `nix` (assuming that it is always required). We can check `$0` to determine how the script was invoked :)

Reply:
Also `s/run.sh/run-e2e.sh/`

Reply:
There is not a great way to determine whether the script has been invoked by `nix develop -c hack/run-e2e.sh` when _inside_ the script.

Best we could do is test an env var which is used when a shell is setup from nix (`IN_NIX_SHELL`). If this variable isn't present, then we should be okay to include `exec nix develop -c "${ROOT_DIR}/hack/run-e2e.sh"` in the script.